app/src/main/java/org/oppia/app/drawer/ExitProfileDialogInterface.kt

```diff
@@ -2,5 +2,6 @@ package org.oppia.app.drawer
 
 /** Interface to handle option selection in [ExitProfileDialogFragment]. */
 interface ExitProfileDialogInterface {
-  fun markHomeMenuCloseDrawer()
+  fun markLastCheckedItemCloseDrawer(lastCheckedItemId: Int, isAdminSelected: Boolean)
+  fun unmarkSwitchProfileItemCloseDrawer()
```

Comment:
We can have multiple approach to solve this problem, but can we investigate here in the first place why we are getting this issue. What's the root cause of this multiple markings rather than solving first.

Reply:
> @prayutsu we need to take a deep look into this why we are having this issue, what sort of things causing this failure and also need to write proper test cases which fails without fix and passes with the fix.

Definitely, so I am sharing my observations. I think the problem is whenever we cancel, the dialog, we call this ```markHomeMenuCloseDrawer```. But we should check the item corresponding to the current fragment. Also if we want this to work when the orientation is changed, we should have the previously checked itemId, so that's why I am passing them in the arguments.

Reply:
Yes, but before the implementation of `markHomeMenuCloseDrawer` this issue was already there. 
We need to a bit deeper look into this, also we might need to check the history of file.

Reply:
Okay, sure